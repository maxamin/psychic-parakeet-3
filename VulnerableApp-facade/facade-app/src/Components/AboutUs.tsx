import React from "react";
import {
  Panel as RSuitePanel,
  PanelGroup as RSuitePanelGroup,
  Icon as RSuiteIcon,
} from "rsuite";

export default class AboutUs extends React.Component {
  Card = () => (
    <div>
      <RSuitePanel
        header="About Us"
        className="VulnerableApp-Facade-AboutUs-Content-Header"
      >
        <div className="VulnerableApp-Facade-AboutUs-Content">
          <p>
            VulnerableApp-Facade was started when we were building{" "}
            <a href="https://github.com/SasanLabs/owasp-zap-fileupload-addon">
              File-upload addon
            </a>{" "}
            for <a href="https://www.zaproxy.org/">Owasp ZAP</a>, which is used
            to find the vulnerabilities in the File Upload functionality. As we
            were developing the addon and writing the scan rules, in order to
            test those scan rules we were using{" "}
            <a href="https://github.com/SasanLabs/VulnerableApp">
              Owasp VulnerableApp
            </a>{" "}
            but as Owasp VulnerableApp is written in Java and is a SpringBoot
            application hence testing scan rules related to PHP, ASP, JSP etc
            are not possible. Hence we thought of building a facade application
            which acts as proxy or gateway to route the requests to vulnerable
            applications written in different tech stacks.
          </p>
        </div>
      </RSuitePanel>
      <RSuitePanel
        header="Contact Us"
        className="VulnerableApp-Facade-AboutUs-Content-Header"
      >
        <p className="VulnerableApp-Facade-AboutUs-Content">
          <RSuiteIcon icon="envelope" role={"img"} /> karan.sasan@owasp.org
        </p>
      </RSuitePanel>
    </div>
  );

  render() {
    return (
      <div>
        <RSuitePanelGroup>
          <this.Card />
        </RSuitePanelGroup>
      </div>
    );
  }
}
