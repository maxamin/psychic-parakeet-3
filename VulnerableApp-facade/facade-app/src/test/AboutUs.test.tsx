import React from "react";
import { render, screen } from "@testing-library/react";

import AboutUs from "./../Components/AboutUs";

describe("About Us", () => {
  it("renders correctly", () => {
    const { container } = render(<AboutUs />);

    expect(container).toMatchSnapshot();
  });

  it("should have a header", () => {
    render(<AboutUs />);

    const header = screen.getAllByRole("rowheader")[0];
    expect(header).toBeInTheDocument();
    expect(header.textContent).toBe("About Us");
  });

  it("should have about us content", () => {
    render(<AboutUs />);

    const hyperLinks = screen.getAllByRole("link");
    const fileUploadAddon = hyperLinks[0];
    expect(fileUploadAddon).toBeInTheDocument();
    expect(fileUploadAddon).toHaveAttribute(
      "href",
      "https://github.com/SasanLabs/owasp-zap-fileupload-addon"
    );
    expect(fileUploadAddon.textContent).toBe("File-upload addon");

    const owaspZap = hyperLinks[1];
    expect(owaspZap).toBeInTheDocument();
    expect(owaspZap).toHaveAttribute("href", "https://www.zaproxy.org/");
    expect(owaspZap.textContent).toBe("Owasp ZAP");

    const vulnerableApp = hyperLinks[2];
    expect(vulnerableApp).toBeInTheDocument();
    expect(vulnerableApp).toHaveAttribute(
      "href",
      "https://github.com/SasanLabs/VulnerableApp"
    );
    expect(vulnerableApp.textContent).toBe("Owasp VulnerableApp");

    expect(
      screen.getByText(/VulnerableApp-Facade was started when we were building/)
    ).toBeInTheDocument();
  });

  it("should have a contact us section", () => {
    render(<AboutUs />);

    const header = screen.getAllByRole("rowheader")[1];
    const icon = screen.getByRole("img");
    const text = screen.getByText("karan.sasan@owasp.org");

    expect(header).toBeInTheDocument();
    expect(header.textContent).toBe("Contact Us");
    expect(icon).toBeInTheDocument();
    expect(text).toBeInTheDocument();
  });
});
