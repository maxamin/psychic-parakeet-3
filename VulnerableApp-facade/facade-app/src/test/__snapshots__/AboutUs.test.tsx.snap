// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Us renders correctly 1`] = `
<div>
  <div>
    <div
      class="rs-panel-group"
    >
      <div>
        <div
          class="VulnerableApp-Facade-AboutUs-Content-Header rs-panel rs-panel-default"
        >
          <div
            class="rs-panel-heading"
            role="rowheader"
            tabindex="-1"
          >
            About Us
          </div>
          <div
            class="rs-panel-body"
          >
            <div
              class="VulnerableApp-Facade-AboutUs-Content"
            >
              <p>
                VulnerableApp-Facade was started when we were building
                 
                <a
                  href="https://github.com/SasanLabs/owasp-zap-fileupload-addon"
                >
                  File-upload addon
                </a>
                 
                for 
                <a
                  href="https://www.zaproxy.org/"
                >
                  Owasp ZAP
                </a>
                , which is used to find the vulnerabilities in the File Upload functionality. As we were developing the addon and writing the scan rules, in order to test those scan rules we were using
                 
                <a
                  href="https://github.com/SasanLabs/VulnerableApp"
                >
                  Owasp VulnerableApp
                </a>
                 
                but as Owasp VulnerableApp is written in Java and is a SpringBoot application hence testing scan rules related to PHP, ASP, JSP etc are not possible. Hence we thought of building a facade application which acts as proxy or gateway to route the requests to vulnerable applications written in different tech stacks.
              </p>
            </div>
          </div>
        </div>
        <div
          class="VulnerableApp-Facade-AboutUs-Content-Header rs-panel rs-panel-default"
        >
          <div
            class="rs-panel-heading"
            role="rowheader"
            tabindex="-1"
          >
            Contact Us
          </div>
          <div
            class="rs-panel-body"
          >
            <p
              class="VulnerableApp-Facade-AboutUs-Content"
            >
              <i
                class="rs-icon rs-icon-envelope"
                role="img"
              />
               karan.sasan@owasp.org
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
